### ISSUE TITLE
Packagist badge returns pre-release version

### ISSUE THREAD

**Issue by hrach** [badges/shields#3609]
Are you experiencing an issue with...

- [x] [shields.io](https://shields.io/#/)
- [ ] My own instance
- [ ] [gh-badges NPM package](https://www.npmjs.com/package/gh-badges)

:beetle: The current stable release of nextras/dbal library is v3.1.0. Badge shows 3.1.0-rc.
- https://github.com/nextras/dbal/releases/tag/v3.1.0
- https://github.com/nextras/dbal/releases/tag/v3.1.0-rc1

:link: https://img.shields.io/packagist/v/nextras/dbal.svg?style=flat

:bulb: /shrug

**Comment by chris48s** [comment]
This is a variation of this issue:

* https://github.com/badges/shields/issues/2190
* https://github.com/badges/shields/issues/1586
* https://github.com/badges/shields/issues/885

tl;dr: When we sort versions from packagist, we assume the versions follow the packagist version schema defined at https://getcomposer.org/doc/04-schema.md#version but packagist doesn't enforce its own schema - it is possible to publish a release which doesn't follow that pattern. If we get a package which has a non-compliant version number, our sort process will produce a spurious result.

**Comment by hrach** [comment]
Hi, thanks for response, I did search for duplicates but somehow missed that, sorry for that.

So, from the mentioned issues: you ignore how packagist/composer world behaves and works today. The opened issue https://github.com/composer/packagist/issues/921 **clearly states** that they are not going to enforce the specs and tools should do a normalization. 

I don't really see [any good in doing this](https://github.com/badges/shields/issues/1586#issuecomment-397822659), it just harms open source users.

**Comment by chris48s** [comment]
> it just harms open source users

How about we take the rhetoric down a notch and start from a position that none of us are giving up our free time to work on open source with malicious intent, shall we..

Nobody's actively working on it, but personally I'm not against accepting a PR to either:

* validate packagist versions and discard invalid versions instead of attempting to sort them or
* attempt to coerce nearly-valid packagist versions before sorting

as its clear this issue is going to come up from time to time. If you're interested in working on it, we can provide some pointers.

**Comment by hrach** [comment]
Thanks for the open thinking. PR created, I've chosen: 
> attempt to coerce nearly-valid packagist versions before sorting
